From 1ae91d8954a8fcd3600b15edcae6fef1d39a3737 Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Tue, 21 Jul 2020 11:05:00 +0800
Subject: [PATCH] Revert "drm/prime: add dmabuf begin/end cpu access"

This reverts commit 4b7491b317285c41f5dde42cad03d0a02f7369f1.
move dmabuf begin/end cpu access to rockchip drm drivers.

Change-Id: Ie143cddf743012768ddae3bb791f720d2933fc52
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/drm_prime.c | 26 --------------------------
 include/drm/drm_drv.h       |  6 ------
 2 files changed, 32 deletions(-)

diff --git a/drivers/gpu/drm/drm_prime.c b/drivers/gpu/drm/drm_prime.c
index 864049556f5c..3b29ac5c18db 100644
--- a/drivers/gpu/drm/drm_prime.c
+++ b/drivers/gpu/drm/drm_prime.c
@@ -490,30 +490,6 @@ int drm_gem_dmabuf_mmap(struct dma_buf *dma_buf, struct vm_area_struct *vma)
 }
 EXPORT_SYMBOL(drm_gem_dmabuf_mmap);
 
-static int drm_gem_dmabuf_begin_cpu_access(struct dma_buf *dma_buf,
-					   enum dma_data_direction dir)
-{
-	struct drm_gem_object *obj = dma_buf->priv;
-	struct drm_device *dev = obj->dev;
-
-	if (!dev->driver->gem_prime_begin_cpu_access)
-		return -EINVAL;
-
-	return dev->driver->gem_prime_begin_cpu_access(obj, dir);
-}
-
-static int drm_gem_dmabuf_end_cpu_access(struct dma_buf *dma_buf,
-					  enum dma_data_direction dir)
-{
-	struct drm_gem_object *obj = dma_buf->priv;
-	struct drm_device *dev = obj->dev;
-
-	if (!dev->driver->gem_prime_end_cpu_access)
-		return -EINVAL;
-
-	return dev->driver->gem_prime_end_cpu_access(obj, dir);
-}
-
 static const struct dma_buf_ops drm_gem_prime_dmabuf_ops =  {
 	.attach = drm_gem_map_attach,
 	.detach = drm_gem_map_detach,
@@ -525,8 +501,6 @@ static const struct dma_buf_ops drm_gem_prime_dmabuf_ops =  {
 	.mmap = drm_gem_dmabuf_mmap,
 	.vmap = drm_gem_dmabuf_vmap,
 	.vunmap = drm_gem_dmabuf_vunmap,
-	.begin_cpu_access = drm_gem_dmabuf_begin_cpu_access,
-	.end_cpu_access = drm_gem_dmabuf_end_cpu_access,
 };
 
 /**
diff --git a/include/drm/drm_drv.h b/include/drm/drm_drv.h
index 151ffaf42d1a..152b3055e9e1 100644
--- a/include/drm/drm_drv.h
+++ b/include/drm/drm_drv.h
@@ -31,7 +31,6 @@
 #include <linux/irqreturn.h>
 
 #include <drm/drm_device.h>
-#include <linux/dma-direction.h>
 
 struct drm_file;
 struct drm_gem_object;
@@ -497,11 +496,6 @@ struct drm_driver {
 	int (*gem_prime_mmap)(struct drm_gem_object *obj,
 				struct vm_area_struct *vma);
 
-	int (*gem_prime_begin_cpu_access)(struct drm_gem_object *obj,
-					  enum dma_data_direction);
-	int (*gem_prime_end_cpu_access)(struct drm_gem_object *obj,
-					 enum dma_data_direction);
-
 	/**
 	 * @dumb_create:
 	 *
-- 
2.35.3

